core/src/library.rs

```diff
@@ -432,14 +452,87 @@ impl<'gc> Library<'gc> {
         self.movie_libraries.keys().collect()
     }
 
+    /// Returns the default Font implementations behind the built in names (ie `_sans`)
+    pub fn default_font(
+        &mut self,
+        name: DefaultFont,
+        ui: &dyn UiBackend,
+        renderer: &mut dyn RenderBackend,
+        gc_context: &Mutation<'gc>,
+    ) -> Vec<Font<'gc>> {
+        // Can't use entry api here as we want to use self for `load_device_font`.
+        // Cache the value as this will be looked up a lot, and font lookup by name can be expensive if lots of fonts exist.
+        if let Some(cache) = self.default_font_cache.get(&name) {
+            return cache.clone();
+        }
+
+        let mut result = vec![];
+        for name in self.default_font_names.entry(name).or_default().clone() {
+            if let Some(font) = self.load_device_font(&name, ui, renderer, gc_context) {
+                result.push(font);
+            }
+        }
+
+        self.default_font_cache.insert(name, result.clone());
+        result
+    }
+
     /// Returns the device font for use when a font is unavailable.
-    pub fn device_font(&self) -> Option<Font<'gc>> {
-        self.device_font
+    pub fn load_device_font(
```

Comment:
minor nit: any chance we can rename one of the `load_device_font`? The duplicate naming adds to the confusion :c

Reply:
Made that rename, good idea!